contracts/oracle/CollateralizationOracle.sol

```diff
@@ -0,0 +1,287 @@
+    /// @param _deposit : the PCVDeposit address to remove from the list.
+    function removeDeposit(address _deposit) external onlyGovernor {
+        // get the token in which the deposit reports its token
+        address _token = deposit2token[_deposit];
+
+        // revert if the deposit is not found
+        require(_token != address(0), "CollateralizationOracle: deposit not found");
+
+        // update maps & arrays for faster access
+        // deposits array for the deposit's token
+        deposit2token[_deposit] = address(0);
+        uint256 _nDepositsWithToken = token2deposits[_token].length;
+        bool found = false;
+        for (uint256 i = 0; !found && i < _nDepositsWithToken; i++) {
+            if (token2deposits[_token][i] == _deposit) {
+                found = true;
+                token2deposits[_token][i] = token2deposits[_token][_nDepositsWithToken - 1];
+            }
+        }
+        token2deposits[_token].pop();
+        // if it was the last deposit to have this token, remove this token from
+        // the arrays also
+        if (token2deposits[_token].length == 0) {
+          isTokenInPcv[_token] = 0;
+          uint256 _nTokensInPcv = tokensInPcv.length;
+          found = false;
+          for (uint256 i = 0; !found && i < _nTokensInPcv; i++) {
+              if (tokensInPcv[i] == _token) {
+                  found = true;
+                  tokensInPcv[i] = tokensInPcv[_nTokensInPcv - 1];
+              }
+          }
+          tokensInPcv.pop();
+        }
+        // remove from the main array
+        uint256 _nDeposits = pcvDeposits.length;
+        found = false;
+        for (uint256 i = 0; !found && i < _nDeposits; i++) {
+            if (pcvDeposits[i] == _deposit) {
+                found = true;
+                pcvDeposits[i] = pcvDeposits[_nDeposits - 1];
+            }
+        }
+        pcvDeposits.pop();
+
+        // emit event
+        emit DepositRemove(msg.sender, _deposit);
+    }
+
+    /// @notice Set the price feed oracle (in USD) for a given asset.
+    /// @param _token : the asset to add price oracle for
+    /// @param _oracle : price feed oracle for the given asset
+    function setOracle(address _token, address _oracle) external onlyGovernor {
+        // add oracle to the map(ERC20Address) => OracleAddress
+        token2oracle[_token] = _oracle;
+
+        // emit event
+        emit OracleSet(msg.sender, _token, _oracle);
+    }
+
+    // ----------- IOracle override methods -----------
+    /// @notice update all oracles required for this oracle to work
+    function update() external override whenNotPaused {
+        for (uint256 i = 0; i < tokensInPcv.length; i++) {
+            IOracle(token2oracle[tokensInPcv[i]]).update();
+        }
+    }
+
+    // @notice returns true if any of the oracles required for this oracle to
+    //         work are outdated.
+    function isOutdated() external override view returns (bool) {
```

Comment:
I think we are going to want to memoize the CR read value, pcv equity etc. Lets not for now, but do some gas analysis once all of the PCV deposits are wired up. For now just have isOutdated return false always. We should use a similar update mechanism to Chainlink i.e. time and threshold based